operator/cmd/syncclusterconfig/sync.go

```diff
@@ -278,29 +287,38 @@ func (s *Syncer) Sync(ctx context.Context, desired map[string]any, usersTXT map[
 	var changed []string
 	// NB: toRemove MUST default to an empty array. Otherwise redpanda will reject our request.
 	removed := []string{}
+	upsert := maps.Clone(desired)
 
 	if s.Mode == SyncerModeDeclarative {
 		// Make cluster config syncing fully declarative.
```

Comment:
mind removing this now outdated comment? Or could you move it onto the `Mode` member to denote the historical behavior of V1 and V2?

Reply:
Yea, I'll tidy these up.